### DISCUSSION TITLE
Why run Android CI on macOS?

### DISCUSSION THREAD

**Discussion by torokati44** [uutils/coreutils#5480] [Q&A] [answered]
I'm looking for ways to run the (quite extensive) test suite of a project that I'm porting to Android (either on GHA or locally), and found your CI setup.
I see that you run the Android checks using `avd` on `macos`.
First I thought that this was to run ARM code (on the M1/M2 runners) without having to emulate that too. But this doesn't seem to be the case, and shouldn't be that big a deal anyway.
So, I'm just interested in why is this done on `macos` instead of `ubuntu`?

**Comment by cakebaker** [comment, marked as answer]
I think it is done on `macos` for performance (and financial) reasons, see https://github.com/ReactiveCircus/android-emulator-runner/issues/46

**Comment by torokati44** [comment]
Ah, I see, thank you so much!
